blog/_posts/2016-10-24-scalafix.md

```diff
@@ -0,0 +1,215 @@
+scala.meta, in the development efforts of these two features.
+
+### Scalac converter
+
+The scalac converter is a key subsystem of scala.meta. For every Scala
+feature, the converter recognizes its representation in the Scala
+compiler and translates it into the corresponding data structures of the
+scala.meta API.
+
+Over the last months, we’ve made great improvements to the converter. We
+have established a comprehensive test suite that consists of a sample of
+over 26.000 Scala source files collected from popular open-source
+projects. Thanks to the joint effort of the team of scala.meta
+[contributors](https://github.com/scalameta/paradise/graphs/contributors),
+the converter now supports about 19.000 source files form the test
+corpus. We are continuously working to increase that number, aiming to
+bring language coverage to 100%.
+
+### Semantic API
+
+The semantic API will enable scala.meta programs to inspect compiler information
+such as type inference, name resolution and other functionality that
+requires a compilation step. This opens opportunities for new scalafix rewrite
+rules that cannot be done on a purely syntactic level.
+
+We already have a
+[working prototype](https://github.com/scalacenter/scalafix/pull/14) 
+of the first scalafix rewrite that uses the semantic API: ExplicitImplicit.
+ExplicitImplicit inserts type annotations for implicit definitions, like this:
+
+~~~scala
+// before
+implicit val x = (_: String).length
+// after
+implicit val x: String => Int = (_: String).length
+~~~
+
+Currently, the ExplicitImplicit rewrite only works for simple cases. In
+order to make the rewrite work in more advanced scenarios, we need to
+further improve the converter as well as the semantic API.
+
+Towards new-style macros
+----------------------------
+
+One could say that by contributing to scala.meta, we get two features
+for the price of one. First, an improved converter and semantic API
+enables us to implement more sophisticated scalafix rewrite rules.
+Secondly, we accelerate the development of a new macro system that will
+work both with scalac and Dotty.
+
+As announced at Scala Days 2016, the current macros based on
+scala.reflect will be going away. Such macros have a number of known
+issues, including over complicated API and lacking IDE support. More
+importantly, the deep coupling between old-style macros and scalac
+means macros written in the old system effectively cannot be
+ported to Dotty. The plan is to discontinue support for scala.reflect
+macros in favor of [a new macro system called
+inline/meta](https://github.com/scala/scala.github.com/pull/567).
+
+A technology preview of the new-style macros came out this summer. This
+early release is still somewhat rough around the edges. Nevertheless,
+[scalafmt](https://olafurpg.github.io/scalafmt/) is already using
+these new macro annotations in production since over a month ago. If you
+are interested to learn more, the best place to get started is 
```

Comment:
I personally avoid gerund-ing when I can, but you're the native speaker ;) Done.